src/WebJobs.Script.WebHost/Security/KeyManagement/KeyVaultSecretsRepository.cs

```diff
@@ -29,24 +25,29 @@ public class KeyVaultSecretsRepository : BaseSecretsRepository
         private const string FunctionKeyPrefix = "host--functionKey--";
         private const string SystemKeyPrefix = "host--systemKey--";
 
-        private readonly Lazy<KeyVaultClient> _keyVaultClient;
-        private readonly string _vaultName;
+        private readonly Lazy<SecretClient> _secretClient;
         private readonly IEnvironment _environment;
 
-        public KeyVaultSecretsRepository(string secretsSentinelFilePath, string vaultName, string connectionString, ILogger logger, IEnvironment environment) : base(secretsSentinelFilePath, logger, environment)
+        public KeyVaultSecretsRepository(string secretsSentinelFilePath, string vaultUri, string clientId, ILogger logger, IEnvironment environment) : base(secretsSentinelFilePath, logger, environment)
         {
-            if (secretsSentinelFilePath == null)
+            if (string.IsNullOrEmpty(secretsSentinelFilePath))
             {
                 throw new ArgumentNullException(nameof(secretsSentinelFilePath));
             }
 
-            _vaultName = vaultName ?? throw new ArgumentNullException(nameof(vaultName));
+            if (string.IsNullOrEmpty(vaultUri))
+            {
+                throw new ArgumentNullException(nameof(vaultUri));
+            }
+
+            Uri keyVaultUri = new Uri(vaultUri);
 
-            _keyVaultClient = new Lazy<KeyVaultClient>(() =>
+            _secretClient = new Lazy<SecretClient>(() =>
             {
-                AzureServiceTokenProvider azureServiceTokenProvider = string.IsNullOrEmpty(connectionString) ? new AzureServiceTokenProvider()
-                    : new AzureServiceTokenProvider(connectionString);
-                return new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback));
+                TokenCredential credential = string.IsNullOrEmpty(clientId) ? new DefaultAzureCredential()
```

Comment:
Switch back to ManagedIdentityCredential

Reply:
Update: after offline discussion the team has decided to go with `DefaultAzureCredential` as there are no real security concerns about enabling all options and the chained approach will lead to more code the team needs to potentially maintain